src/main/java/org/fcrepo/spec/testsuite/authz/WebACDefaultACLs.java

```diff
@@ -0,0 +1,134 @@
+                            final String password) {
+        super(adminUsername, adminPassword, username, password);
+    }
+
+    /**
+     * 5.9-A - Inheritance of ACLs in Fedora implementations is defined by the [SOLIDWEBAC]
+     * ACL Inheritance Algorithm and must be reckoned along the [LDP] containment
+     * relationships linking controlled resources
+     *
+     * @param uri of base fedora uri
+     */
+    @Test(groups = {"MUST"})
+    @Parameters({"param0"})
+    public void aclInheritanceMustUseLdpContainmentRelationships(final String uri) {
+        final TestInfo info =
+            setupTest("5.9-A", "Inheritance of ACLs in Fedora implementations is defined by the [SOLIDWEBAC]" +
+                               "ACL Inheritance Algorithm and must be reckoned along the [LDP] containment " +
+                               "relationships linking controlled resources",
+                      "https://fedora.info/2018/06/25/spec/#inheritance", ps);
+        final String simpleRDFBody = "@prefix dc: <http://purl.org/dc/terms/> \n" +
+                                     "<> dc:title \"Test\".";
+        final Headers headers = new Headers(new Header("Content-Type", "text/turtle"));
+
+        //create a resource
+        final String grandParentUri = createResource(uri, info.getId());
+        //create an acl with acl:default with write privileges
+        createAclForResource(grandParentUri, "user-read-write.ttl", this.username);
+        //create a child  and child acl with read privileges and no acl:default
+        final Response childPost = doPost(grandParentUri, headers, simpleRDFBody);
+        final String childUri = getLocation(childPost);
+        createAclForResource(childUri, "user-read-only-no-default.ttl", this.username);
+
+        //create a grandchild  with no acl.
+        final Response grandChildPost = doPost(childUri, headers, simpleRDFBody);
+        final String grandChildUri = getLocation(grandChildPost);
+
+        // verify that the user can read and write to the grandparent
+        doGet(grandParentUri, false);
+        doPost(grandParentUri, headers, simpleRDFBody, false);
+
+        // verify that the user can read but not write to the child
+        doPostUnverified(childUri, headers, simpleRDFBody, false).then().statusCode(clientErrorRange());
+        doGet(childUri, false);
+
+        //verify that the grandchild can read and write.
+        doGet(grandChildUri, false);
+        doPost(grandChildUri, false);
+    }
+
+    /**
+     * 5.9-B - In the case that the controlled resource is uncontained and has no ACL, or that there is no ACL at any
+     * point in the containment hierarchy of the controlled resource, then the server must supply a default ACL.
+     *
+     * @param uri of base container of Fedora server
+     */
+    @Test(groups = {"SHOULD"})
+    @Parameters({"param0"})
+    public void serverSuppliedDefaultAcl(final String uri) {
```

Comment:
My interpretation is that this test is verifying "controlled resource is uncontained and has no ACL". Should we split out a second test to verify "that there is no ACL at any point in the containment hierarchy of the controlled resource"?

Reply:
I can certainly verify that the root resource is uncontained and has a default ACL.

Reply:
Actually I don't think I can verify this.

Reply:
I can check that the supplied root uri is in fact the root uri.

Reply:
@bbpennel :  what did you have in mind for the second test?

Reply:
Wouldn't the "that there is no ACL at any point in the containment hierarchy of the controlled resource" just mean that you have a non-root container with no explicitly set ACL? The default acl on the root does make things a little confusing, but the child resource would be just as free of having an ACL assigned as the root..

Reply:
I suppose that would make sense.  I'll add it.

Reply:
I suppose what complicates things further is that the default ACL is not specified.  So I'm not sure how I might test that.   I can create a child and verify that it's acl returns a 404.  But I would need to analyze the default acl to determine whether or not I should be able to read it with the non-admin user.